validation/src/main/java/io/micronaut/validation/validator/constraints/ConstraintValidatorContext.java

```diff
@@ -48,5 +48,23 @@
      * @return The root bean under validation.
      */
     @Nullable Object getRootBean();
+
+    /**
+     * Sets a message template to be used for the validation error to override default.
+     *
+     * @param messageTemplateOverride the message to override default
+     */
+    default void overrideMessageTemplate(@Nullable final String messageTemplateOverride) {
```

Comment:
Maybe just call this `messageTemplate`, explicitly prefixing with `override` seems an implementation detail